frontend/src/utilities/__tests__/useWatchImages.spec.ts

```diff
@@ -0,0 +1,117 @@
+import { act } from 'react';
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+import { useWatchImages } from '~/utilities/useWatchImages';
+import { listImageStreams } from '~/api';
+import { mapImageStreamToImageInfo } from '~/utilities/imageStreamUtils';
+import { ImageStreamKind } from '~/k8sTypes';
+import { ImageInfo } from '~/types';
+
+jest.mock('~/api', () => ({
+  listImageStreams: jest.fn(),
+}));
+jest.mock('~/utilities/imageStreamUtils', () => ({
+  mapImageStreamToImageInfo: jest.fn(),
+}));
+jest.mock('~/utilities/useDeepCompareMemoize', () => ({
+  useDeepCompareMemoize: <T>(v: T) => v,
+}));
+
+const listImageStreamsMock = listImageStreams as jest.Mock;
+const mapImageStreamToImageInfoMock = mapImageStreamToImageInfo as jest.Mock;
+
+describe('useWatchImages', () => {
+  const namespace = 'my-namespace';
+  const imageStreams: ImageStreamKind[] = [
+    {
+      metadata: { name: 'jupyter-image', annotations: {}, labels: {} },
+      spec: {},
+      status: {},
+    } as ImageStreamKind,
+  ];
+  const mappedImageInfo: ImageInfo = {
+    name: 'test-imagestream',
+    // eslint-disable-next-line camelcase
+    display_name: 'Test Image',
+    description:
+      'Jupyter notebook image with minimal dependency set to start experimenting with Jupyter environment.',
+    order: 1,
+    dockerImageRepo:
+      'image-registry.openshift-image-registry.svc:5000/opendatahub/jupyter-minimal-notebook',
+    tags: [],
+    url: 'https://github.com//opendatahub-io/notebooks/tree/main/jupyter/minimal',
+    error: '',
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
```

Comment:
please try running the suite w/o this beforeEach, as it might not be strictly required this far down in the testing framework. Removing it will help with performance, as well as potentially removing indirection/obscurity around test case side effects. Thank you!